File: 250315_c++/Graph.cpp
// Graph.cpp
// C++ version of Graph class
#include <iostream>
#include <unordered_map>
#include <vector>
#include <set>
using namespace std;

class Graph {
    unordered_map<int, vector<int>> adjList;
public:
    void addEdge(int u, int v) {
        adjList[u].push_back(v);
        adjList[v].push_back(u);
    }
    vector<int> getNeighbors(int node) const {
        auto it = adjList.find(node);
        if (it != adjList.end()) return it->second;
        return {};
    }
    set<int> getNodes() const {
        set<int> nodes;
        for (const auto& p : adjList) nodes.insert(p.first);
        return nodes;
    }
};

File: 250315/Graph.java
import java.util.*;
public class Graph{
    private Map<Integer,List<Integer>> adjList;
    public Graph(){
        adjList=new HashMap<>();
    }
    public void addEdge(int u,int v){
        adjList.putIfAbsent(u,new ArrayList<>());
        adjList.putIfAbsent(v,new ArrayList<>());
        adjList.get(u).add(v);
        adjList.get(v).add(u);
    }
    public List<Integer> getNeighbors(int node){
        return adjList.getOrDefault(node,new ArrayList<>());
    }
    public Set<Integer> getNodes(){
        return adjList.keySet();
    }
}